### ISSUE TITLE
Ghost Train doors are drawn as closed or fully open, half open is missing

### ISSUE THREAD

**Issue by sr1gh** [OpenRCT2/OpenRCT2#23228]
### Operating System

Windows 10 64-Bit

### OpenRCT2 build

OpenRCT2, v0.4.16 (c1082a3d6c on master) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This bug is a graphical glitch or error

### Describe the issue

The half open state for ghost train landscape doors appears missing. The doors are only shown as opened or closed.

### Steps to reproduce

1. Open a park with a Ghost Train ride that has doors.
2. The doors are not drawn as half open while opening or closing.

**Comment by Gymnasiast** [comment]
The requires the implementation of a map animation, somewhat analogous to what the game does for scenery doors.

**Comment by spacek531** [comment]
Adding a map animation is not an issue in this case. Storing the countdown timer for the two possible doors in the TrackElement is not possible until Entity-Component-System is impemented.

**Comment by Gymnasiast** [comment]
> Adding a map animation is not an issue in this case. Storing the countdown timer for the two possible doors in the TrackElement is not possible until Entity-Component-System is impemented.

Not true. The door state is already saved with 4 bits in the track element, just like it is in RCT1.

**Comment by spacek531** [comment]
Is it not 6 bits? Anyway, I based my assumption on the RCT2 implementation of wall doors, which use 4 or 5 bits, because the animation framerate is half of the tick rate, I believe. It's done in a really strange way that's difficult to decypher. If the RCT1 doors do in fact only need 3 bits to store state then I don't see why we can't dynamically create and destroy map animations to handle the doors. I did so in my track switches tech demo and it was not particularly difficult.

**Comment by Gymnasiast** [comment]
The only states I have ever encountered in a save file are 0 (closed), 2 (half open) and 3 (fully open).

And yes, creating a map animation may not be that difficult. I never claimed it was, though.

**Comment by mixiate** [comment]
Just dropping a couple of notes on this, from looking at RCT1 saves:

Afaict there's 4 door states, closed, closing, opening and open. Those are the first 2 bits. I believe the 3rd bit is like a counter to delay the animation a tick (it possibly also updates on alternate ticks). The opening and closing states both use the same sprite, but the animation update can differentiate between them.
